src/main/java/org/folio/edge/oaipmh/utils/OaiPmhOkapiClient.java

```diff
@@ -2,23 +2,133 @@
+    endpointsMap.put("ListMetadataFormats", "/oai/metadata_formats");
+    endpointsMap.put("ListRecords", "/oai/records");
+    endpointsMap.put("ListSets", "/oai/sets");
+  }
+
   protected OaiPmhOkapiClient(Vertx vertx, String okapiURL, String tenant, long timeout) {
     super(vertx, okapiURL, tenant, timeout);
   }
 
+  /**
+   * This method calls OAI-PMH-MOD to retrieve response for 'verb' from nod-oai-pmh
+   *
+   * @param parameters multimap of HTTP GET parameters
+   * @param headers multimap of HTTP GET headers
+   * @return future with response body
+   */
   public CompletableFuture<String> call(MultiMap parameters, MultiMap headers) {
     VertxCompletableFuture<String> future = new VertxCompletableFuture<>(vertx);
-    //TODO
+    String url = getUrlByVerb(parameters);
+    get(
+      url,
+      tenant,
+      combineHeadersWithDefaults(headers),
+      response -> response.bodyHandler(body -> {
+        int httpStatusCode = response.statusCode();
+        if (httpStatusCode == 200) {
+          String responseBody = body.toString();
+          logger.info(String.format(
+            "Successfully retrieved title info from oai-pmh: (%s) %s",
+            httpStatusCode,
+            responseBody));
+          future.complete(responseBody);
+        } else {
+          String err = String.format(
+            "Failed to get info from oai-pmh: (%s) %s",
+            response.statusCode(),
+            body.toString());
+          logger.error(err);
+          future.complete(body.toString());
+        }
+      }),
+      tr -> {
+        logger.error("Exception in oai-pmh: " + tr.getMessage());
+        future.completeExceptionally(tr);
+      });
     return future;
   }
+
+  /**
+   * This method resolves endpoint for 'verb' from {@link VerbType} based on 'verb' value extracted
+   * from HTTP GET parameters multimap
+   *
+   * @param parameters HTTP GET parameters multimap
+   * @return endpoint for corresponding 'verb'
+   */
+  private String getEndpoint(MultiMap parameters) {
+    String verb = parameters.get(Constants.VERB);
+    String endpoint = endpointsMap.get(verb);
+    // Special processing of GetRecord request with {id} to provide REST-styled call of mod-oai-pmh
+    if (VerbType.GET_RECORD.value().equals(verb)) {
+      try {
+        String identifier = parameters.get(Constants.IDENTIFIER);
+        endpoint = endpoint + "/" + URLEncoder.encode(identifier, URL_ENCODING_TYPE);
+        parameters.remove(Constants.IDENTIFIER);
+      } catch (UnsupportedEncodingException e) {
+        logger.error("Error in identifier encoding " + e.getMessage());
+      }
+    }
+    return endpoint;
+  }
+
+  /**
+   * This method transform MultiMap of HTTP GET request parameters to string representation
+   *
+   * @param parameters HTTP GET parameters multimap
+   * @return string representation of GET request parameters
+   */
+  private String getParametersAsString(MultiMap parameters) {
```

Comment:
I'm pretty sure that you can get this via the VertX RoutingContext object passed to the route handler(s).  E.g. routingCtx.request().query().

Reply:
> ah, good point @epam-pavelk